src/mlpack/methods/dbscan/dbscan_impl.hpp

```diff
@@ -224,18 +226,18 @@ void DBSCAN<RangeSearchType, PointSelectionPolicy>::PointwiseCluster(
  * and also the list of cluster assignments.  This can perform search in batch,
  * naive search).
  */
-template<typename RangeSearchType, typename PointSelectionPolicy>
-template<typename MatType>
-void DBSCAN<RangeSearchType, PointSelectionPolicy>::BatchCluster(
+template<typename RangeSearchType, typename PointSelectionPolicy,
+         typename MatType>
+void DBSCAN<RangeSearchType, PointSelectionPolicy, MatType>::BatchCluster(
     const MatType& data,
     UnionFind& uf)
 {
   // For each point, find the points in epsilon-neighborhood and their distances.
   std::vector<std::vector<size_t>> neighbors;
-  std::vector<std::vector<double>> distances;
+  std::vector<std::vector<ElemType>> distances;
   Log::Info << "Performing range search." << std::endl;
   rangeSearch.Train(data);
-  rangeSearch.Search(Range(0.0, epsilon), neighbors, distances);
+  rangeSearch.Search(RangeType<ElemType>(zero, epsilon), neighbors, distances);
```

Comment:
Try just `ElemType(0)` instead, and I think the `zero` variable won't be needed anymore. :+1:

Reply:
Agreed